### PR TITLE
Use an enum class for ShowOperation::Kind

### PR DESCRIPTION
**Pull request by jez**
### Motivation

I wanted to be able to see what all the values of the `operationName`
were so that I could write code to handle them all in a Vim plugin.

### Test plan

See included automated tests.

### CODE REVIEW

File: main/lsp/ShowOperation.cc [hunk 1]
```diff
@@ -8,16 +8,58 @@ namespace sorbet::realmain::lsp {
...
+            break;
+        case ShowOperation::Kind::SlowPathNonBlocking:
+            return "SlowPathNonBlocking";
+            break;
+        case ShowOperation::Kind::References:
+            return "References";
+            break;
+        case ShowOperation::Kind::SymbolSearch:
+            return "SymbolSearch";
+            break;
```
Comment (jvilk) [inline main/lsp/ShowOperation.cc:35]:
```suggestion
            return "Indexing";
        case ShowOperation::Kind::SlowPathBlocking:
            return "SlowPathBlocking";
        case ShowOperation::Kind::SlowPathNonBlocking:
            return "SlowPathNonBlocking";
        case ShowOperation::Kind::References:
            return "References";
        case ShowOperation::Kind::SymbolSearch:
            return "SymbolSearch";
```

File: main/lsp/ShowOperation.cc [hunk 2]
```diff
@@ -8,16 +8,58 @@ namespace sorbet::realmain::lsp {
...
+        case ShowOperation::Kind::SlowPathNonBlocking:
+            return "Typchecking in background";
+            break;
+        case ShowOperation::Kind::References:
+            return "Finding all references...";
+            break;
+        case ShowOperation::Kind::SymbolSearch:
+            return "Workspace symbol search...";
+            break;
```
Comment (jvilk) [inline main/lsp/ShowOperation.cc:55]:
```suggestion
            return "Indexing files...";
        case ShowOperation::Kind::SlowPathBlocking:
            return "Typechecking...";
        case ShowOperation::Kind::SlowPathNonBlocking:
            return "Typchecking in background";
        case ShowOperation::Kind::References:
            return "Finding all references...";
        case ShowOperation::Kind::SymbolSearch:
            return "Workspace symbol search...";
```

**Review by jvilk — APPROVED** [review]
Had some small nits, otherwise this LGTM.

File: main/lsp/ShowOperation.cc (hunk 2 above)
Comment (jez) [reply main/lsp/ShowOperation.cc:55]: oops, thanks

that was copy paste from when i did this in one `switch` statement in the constructor, but that would have required me to make the fields non-const in order to write into `this->operationName` etc.

**Pull request merged**